var/spack/repos/builtin/packages/bacio/package.py

```diff
@@ -7,13 +7,26 @@
 
 
 class Bacio(CMakePackage):
-    """The bacio ibrary performs binary I/O for the NCEP models, processing
-    unformatted byte-addressable data records, and transforming the little
-    endian files and big endian files."""
+    """The bacio ibrary performs binary I/O for the NCEP models,
+    processing unformatted byte-addressable data records, and
+    transforming the little endian files and big endian files.
+
+    This is part of the NCEPLIBS project.
+
+    """
 
     homepage = "https://noaa-emc.github.io/NCEPLIBS-bacio"
     url      = "https://github.com/NOAA-EMC/NCEPLIBS-bacio/archive/refs/tags/v2.4.1.tar.gz"
 
-    maintainers = ['t-brown']
+    maintainers = ['t-brown', 'edwardhartnett', 'kgerheiser', 'Hang-Lei-NOAA']
 
     version('2.4.1', sha256='7b9b6ba0a288f438bfba6a08b6e47f8133f7dba472a74ac56a5454e2260a7200')
+
+    def setup_run_environment(self, env):
+        env.set('BACIO_LIBd', join_path(self.prefix, 'lib', 'libbacio_8.a'))
```

Comment:
It would be slightly nicer to define these as:
```
env.set('BACIO_LIBd', join_path(self.prefix.lib, 'libbacio_8.a'))
```

Reply:
There is also a `lib64` version too.

Reply:
So, it's `self.prefix.lib` or `self.prefix.lib64`? I thought it dynamically chose the lib dir. It's not known ahead of time which one it'll be.

Reply:
The proper way is to search for libraries: https://spack.readthedocs.io/en/latest/llnl.util.html?highlight=find_libraries#llnl.util.filesystem.find_libraries grep in other packages for examples

Reply:
@edwardhartnett we should use this function to correctly set the path with something like this (not tested):

```
lib4 = find_libraries("libbacio_4.a", self.prefix)
env.set('BACIO_LIB4, lib4)
```